variables.tf

```diff
@@ -37,7 +37,7 @@ variable "badge_enabled" {
 variable "branch" {
   type        = string
   description = "Name of the branch where updates will trigger a build. Used only when `event` is \"branch\""
-  default     = "master"
+  default     = null
```

Comment:
I like the previous change to ensure branch cannot be set to null in modules/branch.  However, I'm not sure if this change makes things better.  With a null default if the top-level module user forgets to set branch then an error will occur when trying to plan.  With a default of "master" an error will not occur if the user does not set branch.  The only way an error could occur is if the user deliberately sets branch to null.

Reply:
Setting this to `null` is so that the default value is only managed in one place, in the branch module. It is possible to make this change only because this PR is also setting `nullable = False` in the branch module... We *can't* set that here at the top-level, because the top-level module needs to support other event types, so this must remain an optional variable.

Reply:
>Setting nullable to false ensures that the variable value will never be null within the module. If nullable is false and the variable has a default value, then Terraform uses the default when a module input argument is null.
>https://www.terraform.io/language/values/variables#disallowing-null-input-values

Reply:
That makes sense...thanks!